client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -844,17 +844,18 @@ Undocumented.prototype.getTitanMailboxAvailability = function ( domain, mailbox,
 /**
  * Get a list of WordPress.com products
  *
- * @param {Function} fn The callback function
+ * @param {object}   query A list of query parameters to include in the request
+ * @param {Function} fn    The callback function
  */
-Undocumented.prototype.getProducts = function ( fn ) {
+Undocumented.prototype.getProducts = function ( query, fn ) {
```

Comment:
This introduces some ambiguity to the `getProducts()` signature - we can now call it in 3 different ways:

* `getProducts( { ... } )`
* `getProducts( () => { ... } )`
* `getProducts( { ... }, () => {} )`

Could we reduce the possible usage patterns to 2? That would make sense to me, because we don't have a scenario where we need to specify both query args and a callback function. In other words, that's what the YAGNI strategy would dictate anyway.

Reply:
Yeah, we can definitely reduce the complexity here. I wrote the method this way mostly to mimic what some others in this same file are doing, but if there's no reasonable use case, I much prefer the idea of cutting out unneeded room for bugs. 🙂